addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/handler/LoxoneMiniserverHandler.java

```diff
@@ -201,16 +211,32 @@ public void channelLinked(ChannelUID channelUID) {
     @Override
     public void initialize() {
         logger.trace("Initializing thing");
+
+        // config.as does not put restrictions on the return time (it is nullable)
+        // but compiler figures out its implementation never returns null and gives a warning
+        // about redundant null check - so we do not check for null here even though this is nullable
+        @Nullable
         LoxoneMiniserverConfig cfg = getConfig().as(LoxoneMiniserverConfig.class);
-        try {
-            InetAddress ip = InetAddress.getByName(cfg.host);
-            server = new LxServer(ip, cfg.port, cfg.user, cfg.password);
-            server.addListener(this);
-            server.update(cfg.firstConDelay, cfg.keepAlivePeriod, cfg.connectErrDelay, cfg.responseTimeout,
-                    cfg.userErrorDelay, cfg.comErrorDelay, cfg.maxBinMsgSize, cfg.maxTextMsgSize);
-            server.start();
-        } catch (UnknownHostException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Unknown host");
+        String host = cfg.host;
+        String user = cfg.user;
+        String password = cfg.password;
+        int port = cfg.port;
+        if (user != null && password != null) {
+            try {
+                @SuppressWarnings("null")
+                @NonNull
+                InetAddress ip = InetAddress.getByName(host);
+                LxServer srv = new LxServer(ip, port, user, password);
+                srv.addListener(this);
+                srv.update(cfg.firstConDelay, cfg.keepAlivePeriod, cfg.connectErrDelay, cfg.responseTimeout,
+                        cfg.userErrorDelay, cfg.comErrorDelay, cfg.maxBinMsgSize, cfg.maxTextMsgSize);
+                srv.start();
+                this.server = srv;
```

Comment:
Simply using `server` from the class instead of local `srv` gives two warnings that server may be null, which is theoretically right. Using a local reference and assigning it to the class variable once all is initialized looked awkward at first to me. But looking from a more purist point of view, I can incline to saying this is actually the right way to code it so it is thread safe. Not that it has to be thread safe here, but as a general practice. 
Although I must say I'm not sure about your remark that 'it suggests that body construct the server might need to be split of', could you explain it more for me?

Reply:
I don't understand my own sentence to, give me some minutes to re-review that part ;-)

Reply:
What I meant as extracting a method or builder or something which builds the complete server given the configuration parts, this method can then return the complete `LxServer` and it can be assigned straight from that method.